bug/60568 Remove permission restriction for users to set their own reminders

akabiru opened the pull request:
# Ticket
<!-- Provide the link to respective work package -->

<!-- Contributors: Please check our PR guide: https://www.openproject.org/docs/development/code-review-guidelines/#preparing-your-pull-request before opening a PR. -->

<!-- Reviewers: Please check our Review guide: https://www.openproject.org/docs/development/code-review-guidelines/#reviewing -->

https://community.openproject.org/work_packages/60568

# What are you trying to accomplish?
<!-- Provide a description of the changes. -->

-  [x] Remove dedicated `manage_own_reminders` project permission and instead rely on `view_work_packages` permission - if a user can see a WP then they can create a reminder on it.
- [x] Ensure Anonymous users are excluded - as anonymous users can view_work_packages in some cases
- [x] Clean up DB of `manage_own_reminders` permission

## Screenshots
<!-- Provide before/after screenshots, videos, or graphs for any visual changes; otherwise, remove this section -->

# What approach did you choose and why?
<!-- This section is a place for you to describe your thought process in making these changes.
     List any tradeoffs you made to take on or pay down tech debt.
     Describe any alternative approaches you considered and why you discarded them. -->

# Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [ ] Tested major browsers (Chrome, Firefox, Edge, ...)

akabiru committed:
[#60568] Remove permission restriction for users to set their own reminders

https://community.openproject.org/work_packages/60568

akabiru committed:
Remove manage own reminders permission

akabiru committed:
Ensure anonymous are restricted from creating reminders

To display reminders, ensure that the user has a `view_work_packages` permission and is logged in.
As anonymous users can view work packages in public projects and OP instances where auth is not required,
this check covers that

machisuji commented on db/migrate/20250117105334_remove_manage_own_reminders_permission.rb:
```diff
@@ -0,0 +1,41 @@
+
+class RemoveManageOwnRemindersPermission < ActiveRecord::Migration[7.1]
+  def up
+    execute <<-SQL.squish
```
👍

machisuji reviewed (approved):
👍 Looks good to me! Thanks for writing the migration in plain SQL. Since this is a simple delete query and no code involved, I'm not worried about this going wrong or adding much time to the migration, either. So there is no need to test the migrations again.

akabiru committed:
Apply semantic order in `isLoggedInAndHasCapabilities$` fn

brunopagno commented on frontend/src/app/features/work-packages/components/wp-details/wp-details-toolbar.component.ts:
```diff
@@ -64,7 +65,9 @@ export class WorkPackageSplitViewToolbarComponent implements OnInit {
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access
-    this.displayReminderButton$ = this.currentUserService.hasCapabilities$('work_package_reminders/modal_body', this.workPackage.project.id);
+    this.displayReminderButton$ = this.currentUserService.isLoggedInAndHasCapabalities$(
+      'work_package_reminders/modal_body',
```
I am probably missing a lot of context, but I find it weird that we have a capability called "modal_body". The name expresses exactly what it is, but it feels more like an implementation choice than an actual "capability".

In any case, I am not expecting any changes, more curious about the context.

akabiru replied:
Good eye, I presume the name was drawn from Controller#action - but it should actually map to a "capability" - in this case "manage". Will update! 👍🏾

akabiru replied:
https://github.com/opf/openproject/pull/17642/commits/c94e902b6ac5e8ce90c33e18c6c580a2824c4fa8
https://github.com/opf/openproject/pull/17642/commits/f6d0fbeaf18ea0fa6dc86d7a5c03ab1711be3148

brunopagno reviewed (approved):
All good from my side :clap:

akabiru committed:
Redefine `work_package_reminders/manage` capability

Use a more semantically correct name, rather than the controller action

akabiru committed:
Just reuse `work_package/read` capability
Managing reminders is based on the `view_work_packages` permission
which is already indentified via `work_packages/read` capability.
It would be redundant to introduce a new capability name

akabiru commented:
CI failures - see https://github.com/opf/openproject/pull/17663 will await confirmation whether to fix in release branch. Merging this now to get QA going
